perception/autoware_map_based_prediction/src/path_generator.cpp

```diff
@@ -210,37 +228,65 @@ PredictedPath PathGenerator::generateStraightPath(
 
 PredictedPath PathGenerator::generatePolynomialPath(
   const TrackedObject & object, const PosePath & ref_path, const double duration,
-  const double lateral_duration, const double speed_limit) const
+  const double lateral_duration, const double backlash_width, const double speed_limit) const
 {
   std::unique_ptr<ScopedTimeTrack> st_ptr;
   if (time_keeper_) st_ptr = std::make_unique<ScopedTimeTrack>(__func__, *time_keeper_);
 
   // Get current Frenet Point
   const double ref_path_len = autoware::motion_utils::calcArcLength(ref_path);
-  const auto current_point = getFrenetPoint(object, ref_path, duration, speed_limit);
+  const auto current_point = getFrenetPoint(object, ref_path.at(0), duration, speed_limit);
 
-  // Step1. Set Target Frenet Point
+  // Step 1. Set Target Frenet Point
   // Note that we do not set position s,
   // since we don't know the target longitudinal position
   FrenetPoint terminal_point;
-  terminal_point.s_vel = current_point.s_vel;
+  terminal_point.s_vel = std::hypot(current_point.s_vel, current_point.d_vel);
   terminal_point.s_acc = 0.0;
-  terminal_point.d = 0.0;
   terminal_point.d_vel = 0.0;
   terminal_point.d_acc = 0.0;
 
-  // Step2. Generate Predicted Path on a Frenet coordinate
+  // calculate terminal d position, based on backlash width
+  {
+    if (backlash_width < 0.01 /*m*/) {
+      terminal_point.d = 0.0;
+    } else {
+      constexpr double return_width = 1.5;  // [m]
+      const double return_coeff =
+        return_width / backlash_width;  // if the object is biased too much, back to the center path
```

Comment:
How this 1.5m decided? What about just use backlash length to switch cases?
Also you'll need to fix `(offset_ratio >= 1 && offset_ratio < return_coeff + 1)` in L265.

If using return_coeff is a better verbalization, I will resolve this comment.

Reply:
This part is to return the target offset to zero when the target lane is a side lane, or the object is outside of any lane.

A. 1.5 m is an half width of the general lane width of 3m. It is expensive or impossible to get proper width of switching lane width. Therefore it is an empirical value.

B. `(offset_ratio >= 1 && offset_ratio < return_coeff + 1)` = `(current_momentum_d >= backlash_width && current_momentum_d  <  backlash_width + return_width )`, which is as my intention. 
The bias will be zero if `current_momentum_d` is outside of the backlash width + the return width.

Reply:
Thanks!

- In my sense, `(current_momentum_d >= backlash_width && current_momentum_d  <  backlash_width + return_width )` is easier to understand.
- In my understandings, `return_width` originally means the half of neighbor lanelet width. Then can we use current lanelet width instead?(But the function do not have access to current lanelet size :face_with_diagonal_mouth: )

Reply:
1. resolved in https://github.com/autowarefoundation/autoware.universe/pull/8602/commits/7b24da2231d9c9647bf781a4dbf2476feffcc366
2. the return width will be calculated as a half of the path width https://github.com/autowarefoundation/autoware.universe/pull/8602/commits/3b85890f7b6bdd9063d0f6d7c747724eedc9a24f 

Test is running https://evaluation.tier4.jp/evaluation/reports/983591ff-2753-5590-a49d-de3163aa7100?project_id=prd_jt